File: app/src/main/java/com/chopcode/rutago/app/models/Rating.java
package com.chopcode.rutago.app.models;

/**
 * Modelo para las calificaciones de conductores
 */
import com.google.firebase.database.IgnoreExtraProperties;
import com.google.firebase.database.PropertyName;

@IgnoreExtraProperties
public class Rating {
    private String id;
    private String passengerId;
    private String passengerName;
    private String driverId;
    private String reservationId;
    private String route;
    private float rating;
    private String comment;
    private long date;

    public Rating() { }

    public Rating(String passengerId, String passengerName, String driverId, 
                  String reservationId, String route, float rating, String comment) {
        this.passengerId = passengerId;
        this.passengerName = passengerName;
        this.driverId = driverId;
        this.reservationId = reservationId;
        this.route = route;
        this.rating = rating;
        this.comment = comment;
        this.date = System.currentTimeMillis();
    }

    // Getters y Setters con mapeo dual
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    @PropertyName("passengerId")
    public String getPassengerId() { return passengerId; }
    @PropertyName("passengerId")
    public void setPassengerId(String passengerId) { this.passengerId = passengerId; }
    @PropertyName("pasajeroId")
    public void setPasajeroId(String pasajeroId) { this.passengerId = pasajeroId; }

    @PropertyName("passengerName")
    public String getPassengerName() { return passengerName; }
    @PropertyName("passengerName")
    public void setPassengerName(String passengerName) { this.passengerName = passengerName; }
    @PropertyName("pasajeroNombre")
    public void setPasajeroNombre(String pasajeroNombre) { this.passengerName = pasajeroNombre; }

    @PropertyName("driverId")
    public String getDriverId() { return driverId; }
    @PropertyName("driverId")
    public void setDriverId(String driverId) { this.driverId = driverId; }
    @PropertyName("conductorId")
    public void setConductorId(String conductorId) { this.driverId = conductorId; }

    @PropertyName("reservationId")
    public String getReservationId() { return reservationId; }
    @PropertyName("reservationId")
    public void setReservationId(String reservationId) { this.reservationId = reservationId; }
    @PropertyName("reservaId")
    public void setReservaId(String reservaId) { this.reservationId = reservaId; }

    public String getRoute() { return route; }
    public void setRoute(String route) { this.route = route; }

    public float getRating() { return rating; }
    public void setRating(float rating) { this.rating = rating; }

    @PropertyName("comment")
    public String getComment() { return comment; }
    @PropertyName("comment")
    public void setComment(String comment) { this.comment = comment; }
    @PropertyName("comentario")
    public void setComentario(String comentario) { this.comment = comentario; }

    @PropertyName("date")
    public long getDate() { return date; }
    @PropertyName("date")
    public void setDate(long date) { this.date = date; }
    @PropertyName("fecha")
    public void setFecha(long fecha) { this.date = fecha; }
}

File: app/src/main/java/com/chopcode/rutago/app/data/models/Rating.kt
package com.chopcode.rutago.app.data.models

import com.google.firebase.database.IgnoreExtraProperties
import com.google.firebase.database.PropertyName

/**
 * 📦 MODEL: Rating
 * Captura el feedback del pasajero tras completar un viaje.
 * Normalizado a Inglés con soporte legacy.
 */
@IgnoreExtraProperties
data class Rating(
    @get:PropertyName("passengerId") @set:PropertyName("passengerId")
    var passengerId: String = "",

    @get:PropertyName("passengerName") @set:PropertyName("passengerName")
    var passengerName: String = "",

    @get:PropertyName("driverId") @set:PropertyName("driverId")
    var driverId: String = "",

    @get:PropertyName("reservationId") @set:PropertyName("reservationId")
    var reservationId: String = "",

    @get:PropertyName("route") @set:PropertyName("route")
    var route: String = "",

    @get:PropertyName("rating") @set:PropertyName("rating")
    var rating: Float = 0.0f,

    @get:PropertyName("comment") @set:PropertyName("comment")
    var comment: String = "",

    @get:PropertyName("date") @set:PropertyName("date")
    var date: Long = System.currentTimeMillis()
) {
    @get:PropertyName("id") @set:PropertyName("id")
    var id: String? = null

    // =========================================================================
    // 🌍 LEGACY COMPATIBILITY
    // =========================================================================

    @get:PropertyName("pasajeroId") @set:PropertyName("pasajeroId")
    var pasajeroIdLegacy: String
        get() = passengerId
        set(value) { passengerId = value }

    @get:PropertyName("pasajeroNombre") @set:PropertyName("pasajeroNombre")
    var pasajeroNombreLegacy: String
        get() = passengerName
        set(value) { passengerName = value }

    @get:PropertyName("conductorId") @set:PropertyName("conductorId")
    var conductorIdLegacy: String
        get() = driverId
        set(value) { driverId = value }

    @get:PropertyName("reservaId") @set:PropertyName("reservaId")
    var reservaIdLegacy: String
        get() = reservationId
        set(value) { reservationId = value }

    @get:PropertyName("comentario") @set:PropertyName("comentario")
    var comentarioLegacy: String
        get() = comment
        set(value) { comment = value }

    @get:PropertyName("fecha") @set:PropertyName("fecha")
    var fechaLegacy: Long
        get() = date
        set(value) { date = value }
}
